src/libfetchers/tarball.cc

```diff
@@ -107,15 +107,16 @@ DownloadFileResult downloadFile(
 }
 
 static DownloadTarballResult downloadTarball_(
-    const Settings & settings, const std::string & url, const Headers & headers, const std::string & displayPrefix)
+    const Settings & settings, const std::string & urlS, const Headers & headers, const std::string & displayPrefix)
 {
+    auto url = parseURL(urlS);
 
     // Some friendly error messages for common mistakes.
     // Namely lets catch when the url is a local file path, but
     // it is not in fact a tarball.
-    if (url.rfind("file://", 0) == 0) {
+    if (url.scheme == "file") {
         // Remove "file://" prefix to get the local file path
-        std::string localPath = url.substr(7);
+        std::string localPath = renderUrlPathEnsureNoSlash(url.path);
```

Comment:
```suggestion
        std::filesystem::path localPath = renderUrlPathEnsureNoSlash(url.path);
```
I think that one will not be more work :)